dictionary_web/script.js

```diff
@@ -0,0 +1,165 @@
+// DOM elements
+const darkModeToggle = document.getElementById("darkModeToggle");
+const fontFamilySelector = document.getElementById("fontFamily");
+const loadingSpinner = document.getElementById("loadingSpinner");
+const searchInput = document.getElementById("searchInput");
+const resultsContainer = document.querySelector(".results-container");
+const sourceContainer = document.querySelector(".source-container");
+const mainContent = document.querySelector(".main");
+const emptySearchInputMessage = document.querySelector(
+  ".empty-search-field-msg"
+);
+
+// Dark Mode Toggle
+darkModeToggle.addEventListener("change", (e) => {
+  if (e.target.checked) {
+    document.body.classList.add("dark-mode");
+  } else {
+    document.body.classList.remove("dark-mode");
+  }
+});
+
+// Event listener for font family change
+fontFamilySelector.addEventListener("change", (e) => {
+  const selectedFont = e.target.value;
+  switch (selectedFont) {
+    case "Sans Serif":
+      document.body.style.fontFamily = "sans-serif";
+      break;
+    case "Serif":
+      document.body.style.fontFamily = "serif";
+      break;
+    case "Mono":
+      document.body.style.fontFamily = "monospace";
+      break;
+    default:
+      document.body.style.fontFamily = "sans-serif";
+  }
+});
+
+searchInput.addEventListener("keypress", (e) => {
+  const word = searchInput.value.trim();
+  if (e.key === "Enter") {
```

Comment:
Does this implementation mean you have to type out the full word before you click enter? usually it is recommended that you begin searching immediately there is a search input. you can look into that

Reply:
I'm working on it to begin searching immediately if there is a search input